app/utils/callback/lava.py

```diff
@@ -60,17 +85,67 @@
 }
 
 
-def _get_definition_meta(meta, job_data):
+def _get_test_case_data(meta, tc_data, job_data, META_DATA_MAP):
+    """Parse the test definition data from the test suite name
+
+    Parse the data from the LAVA v2 job definition sent with the callback
+    and populate the required fields to store in the database.
+
+    :param meta: The test case data to populate.
+    :type meta: dictionary
+    :param tc_data: The JSON data from the callback.
+    :type tc_data: dict
+    :param job_data: The map of keys to search for in the JSON and update.
+    :type job_data: dict
+    :param META_DATA_MAP: The dict of keys to parse and add in the data.
+    :type META_DATA_MAP: list
+    """
+    test_key = None
+    common_meta = {
+        "version": "1.0",
+        # Time shoud be passed in the result as well
+        "time": "0.0",
+    }
+    # TODO: Fix ?
+    # Kind of workaround because the key for a test will be <number>_test_name
+    # Always using first found key for now
+    for key in job_data["results"]:
+        if meta[models.NAME_KEY] in key:
+            test_key = key
+            break
+
+    if test_key is None:
+        utils.LOG.error("Could not find test data for '%s' in lava callback",
+                        meta[models.NAME_KEY])
+        return
+
+    tests = yaml.load(job_data["results"][test_key],
+                      Loader=yaml.CLoader)
+    for test in tests:
+        test_case = {x: test[y] for x, y in META_DATA_MAP.iteritems()}
+        test_case.update(common_meta)
+        # If no set is defined make it part of a generic one
+        if "set" in test["metadata"]:
```

Comment:
Finding the test sets from the test case data. Done this way because lava callback contains a test suite with multiple test cases. Each test case might a test set name. If no test set is defined a default one is used